plugins/auth-backend/src/identity/router.ts

```diff
@@ -68,7 +72,40 @@ export function bindOidcRouter(
     res.status(501).send('Not Implemented');
   });
 
-  router.get('/v1/userinfo', (_req, res) => {
-    res.status(501).send('Not Implemented');
+  // This endpoint doesn't use the regular HttpAuthService, since the contract
+  // is specifically for the header to be communicated in the Authorization
+  // header, regardless of token type
+  router.get('/v1/userinfo', async (req, res) => {
+    const token = req.headers.authorization?.replace(/^Bearer\s+/i, '');
+    if (!token) {
+      throw new AuthenticationError('No token provided');
+    }
+
+    const credentials = await auth.authenticate(token, {
+      allowLimitedAccess: true,
+    });
+
+    if (!auth.isPrincipal(credentials, 'user')) {
+      throw new InputError(
+        'Userinfo endpoint must be called with user credentials',
```

Comment:
Is this the wording we use, or "with a user principal"? I'm happy either way, but the incoming thing from a service could be seen as "service credentials" that carry a user principal. Not sure if one or the other of those wordings make it sound more like "with a user token" which would give the wrong signals